compiler/rustc_ty_utils/src/instance.rs

```diff
@@ -90,13 +92,30 @@ fn resolve_instance_raw<'tcx>(
             let ty = args.type_at(0);
             ty::InstanceKind::AsyncDropGlue(def_id, ty)
         } else {
-            debug!(" => free item");
+            debug!(" => free or inherent item");
             ty::InstanceKind::Item(def_id)
         };
 
         Ok(Some(Instance { def, args }))
     };
-    debug!("resolve_instance: result={:?}", result);
+    if let Ok(Some(Instance { def: ty::InstanceKind::Item(def_id), args })) = result
+        && matches!(tcx.def_kind(def_id), DefKind::Const | DefKind::AssocConst)
+    {
+        debug!(" => resolved to const item");
+        let ct = tcx.const_of_item(def_id).instantiate(tcx, args);
```

Comment:
this probably is also wrong for the same reason :thinking: We ought to figure out a better way to get the actually evaluatable thing of const items for const eval